### PR TITLE
Slime's End Cliffs

### PR DESCRIPTION
**Pull request by aristotlepenguin**
A new subzone on the map: shortened version is sec
command list:
!toss <item> :
    Throw an item off the cliff. Thrown items will either sink into the sea, or get eaten by sea monsters. 
    These creatures kick up slime when they chow down, so make sure to !scavenge it.
!jump
    A form of suicide.
    All your items will be discarded as though they were !tossed. However, adorned and equipped 
    weapons should not delete. You'll leave a portion of your slime on the cliff when the krakens 
    consume you.
!push <player>
    Like !jump, but for other players.
    Players can't be !pushed if they have the "Fat Chance" or "Big Bones" mutation.
    However, having "Light as a Feather" overrides both of these and makes you vulnerable to 
    pushing.
    !bully is an alternate command. 
    !push and !bully also have a dual function outside of the Cliffs.

### CODE REVIEW

File: ewcfg.py [hunk 1]
```diff
@@ -5749,6 +5761,20 @@ def atf_armcannon(ctn = None):
...
+"You decide to give {} a slime swirly in a nearby puddle. It's so shallow that they mostly get a faceful of gravel.",
+"You tie {} to a tree and slap them around senselessly. You untie them once their face and belly bruise cherry red.",
+"You flag down a muscle car on the road and shout: \"HEY! {} FUCKED YOUR WIFE!\" The good man parks on the side of the road and starts beating the everloving shit out them. -SLIMEOID- cowers in the corner, now scarred for life and afraid of dads.",
+"You pull on {}'s hair, ripping some out and causing them to cry. They should fucking grow up.",
+"You reach into {}'s shirt and give them a purple nurple. Man, these bullying tactics are getting kind of gay.",
+"You whip out your dick and pee on {}'s wife. Fuck. That's a power move right there.",
+"You scream \"HEY {}! NICE -COSMETIC- YOU'RE WEARING! DID YOUR MOM BUY IT FOR YA?\"",
+"You grab -SLIMEOID- and give them a noogie. Just when {} thinks this is all fun and games, you throw -SLIMEOID- into the street. They have a panic attack trying to get past all the traffic and back to safety."
```
Comment (magicksRampage) [inline ewcfg.py]: you can get the same result using {slimeoid} and format(slimeoid = slimeoid.name)

File: ewcmd.py [hunk 2]
```diff
@@ -838,3 +839,208 @@ def weather_txt(id_server):
...
+
+		cliff_inventory = ewitem.inventory(id_server=cmd.message.server.id, id_user=targetmodel.id_user)
+		for item in cliff_inventory:
+			item_object = ewitem.EwItem(id_item=item.get('id_item'))
+			if item.get('soulbound') == True:
+				pass
+
+			elif item_object.item_type == ewcfg.it_weapon:
+				if item.get('id_item') == targetmodel.weapon:
+					ewitem.give_item(id_item=item_object.id_item, id_user="ocean", id_server=cmd.message.server.id)
```
Comment (magicksRampage) [inline ewcmd.py]: it won't work unless you use ewcfg.poi_id_slimesea

File: ewcmd.py [hunk 3]
```diff
@@ -838,3 +839,208 @@ def weather_txt(id_server):
...
+				elif slimeoid_model == "":
+					response = "You push {} into a puddle of sludge, laughing at how hopelessly dirty they are.".format(target.display_name)
+
+		else:
+			response = "You can't bully them because they aren't here."
+
+	elif user_data.life_state == ewcfg.life_state_corpse:
+		response = "You attempt to push {} off the cliff, but your hand passes through them. If you're going to push someone, make sure you're corporeal.".format(target.display_name)
+
+	elif targetmodel.life_state == ewcfg.life_state_corpse:
```
Comment (magicksRampage) [inline ewcmd.py:900]: kingpins also need to be immune

File: ewcmd.py [hunk 4]
```diff
@@ -838,3 +839,208 @@ def weather_txt(id_server):
...
+			await ewutils.send_message(cmd.client, sewerchannel, deathreport)
+
+	return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
+
+async def jump(cmd):
+	user_data = EwUser(member=cmd.message.author)
+
+	if user_data.poi != ewcfg.poi_id_slimesendcliffs:
+		response = "You jump. Nope. Still not good at parkour."
+	elif user_data.life_state == ewcfg.life_state_corpse:
```
Comment (magicksRampage) [inline ewcmd.py:961]: kingpins need to be immune to this, too

File: ewcmd.py [hunk 5]
```diff
@@ -838,3 +839,208 @@ def weather_txt(id_server):
...
+		response = "You can't push more than one person at a time."
+		return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
+
+	target = cmd.mentions[0]
+	targetmodel = EwUser(member=target)
+	target_mutations = targetmodel.get_mutations()
+	user_mutations = user_data.get_mutations()
+
+	server = cmd.message.server
+	member_object = server.get_member(targetmodel.id_user)
```
Comment (magicksRampage) [inline ewcmd.py]: this is the same as the 'target' variable

File: ewcmd.py [hunk 6]
```diff
@@ -838,3 +839,208 @@ def weather_txt(id_server):
...
+		response = "You attempt to push {} off the cliff, but your hand passes through them. If you're going to push someone, make sure you're corporeal.".format(target.display_name)
+
+	elif targetmodel.life_state == ewcfg.life_state_corpse:
+		response = "You try to give ol' {} a shove, but they're a bit too dead to be taking up physical space.".format(target.display_name)
+
+	elif (ewcfg.mutation_id_bigbones in target_mutations or ewcfg.mutation_id_fatchance in target_mutations) and ewcfg.mutation_id_lightasafeather not in target_mutations:
+		response = "You try to push {}, but they're way too heavy. It's always fat people, constantly trying to prevent your murderous schemes.".format(target.display_name)
+
+	elif ewcfg.mutation_id_lightasafeather in target_mutations:
+		response = "You pick {} up with your thumb and index finger, and gently toss them off the cliff. Wow. That was easy.".format(target.display_name)
```
Comment (magicksRampage) [inline ewcmd.py]: why doesn't all the stuff in the else case happen here?

File: ewcmd.py [hunk 7]
```diff
@@ -838,3 +839,208 @@ def weather_txt(id_server):
...
+	user_data = EwUser(member=cmd.message.author)
+	item_search = ewutils.flattenTokenListToString(cmd.tokens[1:])
+	item_sought = ewitem.find_item(item_search=item_search, id_user=cmd.message.author.id, id_server=user_data.id_server)
+
+	if user_data.poi != ewcfg.poi_id_slimesendcliffs:
+		return await ewitem.discard(cmd=cmd)
+
+	elif item_sought:
+		item_obj = EwItem(id_item=item_sought.get('id_item'))
+		if item_obj.soulbound == True:
```
Comment (magicksRampage) [inline ewcmd.py:1009]: also make it so you can't toss your weapon, if you're married to it

File: ewcmd.py [hunk 8]
```diff
@@ -838,3 +839,208 @@ def weather_txt(id_server):
...
+	districtmodel = ewdistrict.EwDistrict(id_server=id_server, district=ewcfg.poi_id_slimesendcliffs)
+	slimetotal = 0
+
+	if random.randrange(500) < 125 or item_obj.item_type == ewcfg.it_questitem or item_obj.item_type == ewcfg.it_medal or item_obj.item_props.get('rarity') == ewcfg.rarity_princeps or item_obj.item_props.get('id_cosmetic') == "soul" or item_obj.item_props.get('id_furniture') == "propstand":
+		response = "You toss the {} off the cliff. It sinks into the ooze disappointingly.".format(item_name)
+		ewitem.give_item(id_item=id_item, id_server=id_server, id_user=ewcfg.poi_id_slimesea)
+
+	elif random.randrange(500) < 498:
+		response = "You toss the {} off the cliff. A nearby kraken swoops in and chomps it down with the cephalapod's equivalent of a smile. Your new friend kicks up some sea slime for you. Sick!".format(item_name)
+		slimetotal = 10000 + random.randrange(50000)
```
Comment (magicksRampage) [inline ewcmd.py]: this is way too much slime for a single item. people will be dumping glasses of water from the speakeasy here en masse

**Review by magicksRampage — CHANGES_REQUESTED** [review]

File: ewcmd.py (hunk 2 above)
Comment (aristotlepenguin) [reply ewcmd.py]: sorry about that, was using a different id before miller made slimesea fishing a thing

**Pull request merged**